spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/EncryptionController.java

```diff
@@ -102,23 +102,33 @@ public String getPublicKey() {
 
 	}
 
-	@RequestMapping(value = "/key", method = RequestMethod.POST, params = { "!password" })
+    @RequestMapping(value = "/key", method = RequestMethod.POST, params = { "!password" })
 	public ResponseEntity<Map<String, Object>> uploadKey(@RequestBody String data,
 			@RequestHeader("Content-Type") MediaType type) {
 
 		Map<String, Object> body = new HashMap<String, Object>();
 		body.put("status", "OK");
 
-		encryptor = new EncryptorFactory().create(stripFormData(data, type, false));
-
+		TextEncryptor encryptor = new EncryptorFactory().create(stripFormData(data, type, false));
+        updateEncryptor(encryptor);
 		if (encryptor instanceof RsaKeyHolder) {
 			body.put("publicKey", ((RsaKeyHolder) encryptor).getPublicKey());
 		}
 		logger.info("Key changed with literal value");
 		return new ResponseEntity<Map<String, Object>>(body, HttpStatus.CREATED);
-
 	}
 
+    // this is temporary solution to support existing REST API
+    // downcasting is only necessary until we introduce some key management abstraction,
+    // we don't want TextEncryptorLocator to have setEncryptor method
+    private void updateEncryptor(TextEncryptor encryptor) {
```

Comment:
Could you elaborate on this? EncryptorFactory has been moved recently to spring-cloud-context module and does not have any state. Instead of changing the state on SingleTextEncryptorLocator I would rather add new key to some key storage or something similar but this seems to be out of scope for this PR.